FinniversKit/Sources/SwiftUI Components/SwiftUISelectionListView.swift

```diff
@@ -0,0 +1,101 @@
+import SwiftUI
+
+public struct SwiftUISelectionListView<ItemValue>: View {
+    public var items: [SwiftUISelectionListItemModel<ItemValue>]
+    @Binding public var selectedItem: ItemValue?
+
+    @State private var selectedItemIndex: Int?
+    private var cornerRadius: CGFloat = 5
+
+    public init(items: [SwiftUISelectionListItemModel<ItemValue>], selectedItem: Binding<ItemValue?>) {
+        self.items = items
+        self._selectedItem = selectedItem
+    }
+
+    public var body: some View {
+        VStack(spacing: 0) {
+            ForEach(0..<items.count, id: \.self) { itemIndex in
```

Comment:
There'a a lot of index handling in this view. Isn't it possible to iterate through item in items here instead? So we can just check if item is last/first in list, or if `item == selectedItem`, `item.isSelected =`, etc.?
Feels a bit more robust. But might be some blocker for doing this that I don't see just by reading 😄

Reply:
I think I tried that, but for handling `isSelected` changes and animations we need that index anyways.